File: CCC-school-work/comporg/pnc2/mchon/ruby/primeregbs.rb
#!/usr/bin/ruby

require 'benchmark'
	#################################################################
	##
	## Error checking
	##

	#################################################################
	##
	## We are using benchmark to measure the time
	##
	time = Benchmark.measure {
	#################################################################
	##
	## Variable declaration
	##
	max=Integer(ARGV[0])
	flag=0
	number=2
	qty=0

	#################################################################
	##
	## Main algorithm
	##
	#################################################################
	##
	## This loop will iterate until we print out the right amount
	## of prime number
	##
	while qty < max do
		#############################################################
		##
		## Uses the brute force method to check for primality
		##
		for factor in 2..Math.sqrt(number) do
			#########################################################
			##
			## Checks to see if there is a factor. If there is a
			## factor, flag that it is a composite number
			##
			if number%factor == 0 then
				flag = 1
				break
			end
		end
		#############################################################
		##
		##  
		##
		if flag == 0 then
			STDERR.print number
			STDERR.print " "
			qty += 1
		end
		flag = 0
		number += 1
	end
	STDERR.puts ""
}
#####################################################################
##
## Prints out the time
##
puts "%.6f" % time.real

File: CCC-school-work/comporg/pnc2/mchon/java/primeregbs.java

public class primeregbs
{
	public static void main(String[] args)
	{
		/////////////////////////////////////////////////////////////////////
		//
		// Error checking
		//
		if (args.length < 1)
		{
			System.err.println("Insufficient argument!");
			System.exit(0);

		}
		long start = System.currentTimeMillis();
		/////////////////////////////////////////////////////////////////////
		//
		// Variable declaration
		//
		int max 	= Integer.parseInt(args[0]);
		char flag 	= 0;
		int factor 	= 0;
		int number 	= 2;
		int qty 	= 0;
		/////////////////////////////////////////////////////////////////////
		//
		// Main algorithm
		//

		/////////////////////////////////////////////////////////////////////
		//
		// Continuously checks until all the qty are print out
		//
		while (qty < max)
		{
			/////////////////////////////////////////////////////////////////
			//
			// Starting from 2 up to the current number, checks to see if
			// there are any factor in the number
			//
			for (factor = 2; factor <= Math.sqrt(number); factor++)
			{
				/////////////////////////////////////////////////////////////
				//
				// Checks to see if there are factors. If the number has
				// a factor other than 1 and itself, flag it as composite
				//
				if (number%factor == 0)
				{
					flag = 1;
					break;
				}
			}
			/////////////////////////////////////////////////////////////////
			//
			//	If the flag wasnt set off, then its a prime so print it out
			//
			if (flag == 0)
			{
				System.err.printf(number + " ");
				qty++;
			}
			number++;
			flag = 0;
		}
		System.err.println();
		long endTime = System.currentTimeMillis();
		float deltaTime = ( endTime -  start);
		System.out.printf("%8.6f\n",deltaTime/1000);
	}
}
